src/data-access/repositories/mongo/mongo-part-changed-listener.ts

```diff
@@ -0,0 +1,84 @@
+import { BaseMongoRepository } from './base-mongo-repository'
+import { DataChangedListener } from '../interfaces/data-changed-listener'
+import { MongoDatabase } from './mongo-database'
+import { MongoEntityConverter, MongoPart, MongoSegment } from './mongo-entity-converter'
+import {
+  ChangeStream,
+  ChangeStreamDeleteDocument,
+  ChangeStreamDocument,
+  ChangeStreamInsertDocument,
+  ChangeStreamReplaceDocument
+} from 'mongodb'
+import { MongoChangeEvent } from './mongo-enums'
+import { Part } from '../../../model/entities/part'
+import { PartRepository } from '../interfaces/part-repository'
+
+const PART_COLLECTION_NAME: string = 'parts'
+
+export class MongoPartChangedListener extends BaseMongoRepository implements DataChangedListener<Part> {
+
+  private onCreatedCallback: (part: Part) => void
+  private onUpdatedCallback: (part: Part) => void
+  private onDeletedCallback: (partId: string) => void
+
+  constructor(
+    mongoDatabase: MongoDatabase,
+    mongoEntityConverter: MongoEntityConverter,
+    private readonly partRepository: PartRepository
+  ) {
+    super(mongoDatabase, mongoEntityConverter)
+    mongoDatabase.onConnect(PART_COLLECTION_NAME, () => this.listenForChanges())
+  }
+
+  private listenForChanges(): void {
+    const options = { fullDocument: 'updateLookup' }
```

Comment:
Do we have a type for `options`?

Reply:
We do: `ChangeStreamOptions`
It's been updated.